core/src/main/java/hudson/model/queue/WorkUnit.java

```diff
@@ -85,6 +86,10 @@ public Executable getExecutable() {
     @Restricted(NoExternalUse.class)
     public void setExecutable(Executable executable) {
         this.executable = executable;
+        if (executable instanceof Run) {
+            // TODO: Not possible via the QueueListener it seems. Is there something else?
```

Comment:
(But again since we are inside core code it is more straightforward to do it like you did.)